protocols/rendezvous/src/codec.rs

```diff
@@ -256,11 +256,12 @@ impl libp2p_request_response::Codec for Codec {
     where
         T: AsyncRead + Unpin + Send,
     {
-        if let Some(result) = FramedRead::new(io, RendezvousCodec::default()).next().await {
-            return Ok(result?);
-        }
+        let message = FramedRead::new(io, RendezvousCodec::default())
+            .next()
+            .await
+            .ok_or(io::ErrorKind::UnexpectedEof)??;
 
-        Err(io::ErrorKind::InvalidInput.into())
+        Ok(message)
```

Comment:
Another idea for follow-up work: Similarly as you did it in https://github.com/libp2p/rust-libp2p/pull/3952, we could add a generic `Codec` implementation that can be created from an `asynchronous_codec::Encoder` and `asynchronous_codec::Decoder`.

Just need to find a good name for it ...

Reply:
I removed `RendezvousCodec` and implemented `Decoder`/`Encoder` for `Codec`.
Looks good for me ))